lib/screens/v2/widget_instance/elevator_status.ex

```diff
@@ -188,6 +199,24 @@ defmodule Screens.V2.WidgetInstance.ElevatorStatus do
     )
   end
 
+  # The next 4 functions are almost identical. Worth consolidating? Esp because the above functions
+  # are already named in a descriptive way
+
+  # @spec alert_is_relevant?(Alert.t(), DateTime.t(), atom()) :: boolean()
+  # defp alert_is_relevant?(alert, now, option) do
+    # parent_station_id = parent_station_id(t)
+
+    # active = Alert.happening_now?(alert, now)
+
+    # case option do 
+    #   :upcoming -> not active and Enum.any?(entities, fn e -> e.stop == parent_station_id end)
+    #   :here -> active and Enum.any?(entities, fn e -> e.stop == parent_station_id end)
+    #   :connecting -> more complicated logic OR run BaseAlert.location === :upstream / :downstream
+    #   :elsewhere -> 
+    #     get the parent stations for the entities
+    #     active and Enum.any?(parent_stations, fn station != parent_station_id end)
+  # end
```

Comment:
We might be able to do something like this, although it would need to support multiple options unioned together, so that we can express "is this upcoming _and_ here?" for example.

We would need to be careful that we're not just duplicating `BaseAlert.location` for parts of this.

Overall, "relevance" seems much more per-screen-type and even per-widget-type, though. I don't think a high-level concept like this would fit well in a module that contains simpler utility functions. It would be better to use those simple functions to build different versions of this (which is kind of what we're already doing) for all of our different definitions of "relevant".

Reply:
I think the function I stubbed out here handles the unioning of options, unless I'm misunderstanding your comment?

True, it's not useful to rehash BaseAlert.location, but the location logic of this function is super simple. Here, not here, and downstream (but BaseAlert.location is used for downstream). BaseAlert is way more complicated.

As for the overall statement about relevance, agreed! 👍